pkg/apis/autoscaling/v1alpha1/metricsproducer_validation.go

```diff
@@ -12,9 +12,113 @@ See the License for the specific language governing permissions and
 limitations under the License.
 */
 
+// +kubebuilder:webhook:verbs=create;update,path=/validate-autoscaling-karpenter-sh-v1alpha1-metricsproducer,mutating=false,sideEffects=None,failurePolicy=fail,groups=autoscaling.karpenter.sh,resources=metricsproducers,versions=v1alpha1,name=vmetricsproducer.kb.io
 package v1alpha1
 
-import "fmt"
+import (
+	"fmt"
+	"k8s.io/apimachinery/pkg/runtime"
+	"regexp"
+	"strings"
+	"time"
+)
+
+// Only Validates ScheduledCapacity MetricsProducer right now
+func (m *MetricsProducer) ValidateCreate() error {
+	if m.Spec.Schedule != nil {
+		if err := m.ValidateSchedule(); err != nil {
+			return err
```

Comment:
why not just return directly like on 34?

Reply:
You make a good point.